### ISSUE TITLE
The same letter multiple times is considered a correct word?

### ISSUE THREAD

**Issue by salgiza** [streetsidesoftware/vscode-spell-checker#760]
Ok, I just lost a few minutes of my life because I thought that the spellchecker was not working and, in order to test it, I just pressed the 'x' key a few times: xxxxx... which did not lead to any spelling errors.

After a few more tests, I noticed that it was actually working, it's just that if I write 'x', 'xxx', 'xxxxxx', 'xxxxxxx', it doesn't consider it an error. Same thing with any other letter:

![image](https://user-images.githubusercontent.com/6630306/111451073-57abc200-8711-11eb-9b30-35a750651fce.png)

(Why sdf is considered a word, I don't know either, by the way)

Anyway, now that I know it I guess it is not a big deal, but I'm not sure whether this is intended or a bug, so I thought that I should let you know.

Thanks for a great extension, by the way!

**Comment by Jason3S** [comment]
@salgiza,

The spell checker ignores repeated letters because it is quite common in code.

It also ignores short words, because they are also often just variables. This can be configured:

![image](https://user-images.githubusercontent.com/3740137/111456521-66957300-8717-11eb-8860-015b16609820.png)

**Comment by salgiza** [comment]
That makes sense! But then, maybe as an improvement suggestion:

* Would it be possible to set that value depending on the file type? If I'm writing a markdown file I actually want the spellchecker to warn me of any misspelling errors, regardless of word length (but I'd like to keep that value for .cs files, for example).
* Having that parameter, what is the point of ignoring "xxxxxxxxxx", for example? I mean, I've declared my fair share of "ii" and "zz"... variables. But is there any language where it is common to declare a variable such as "xxxxxx" or "sssss"? (Maybe there is, I'm actually asking)

**Comment by Jason3S** [comment]
It is possible to change the behavior on a per file basis using overrides:

```javascript
"cSpell.overrides": [
    {
        "filename": ["*.md", "*.tex"],
        "minWordLength": 3, // going lower than 3 doesn't make a lot of sense due to all the acronyms.
    }
]
```

If you want to forbid common mistakes, you can have the spell checker flag them:
![image](https://user-images.githubusercontent.com/3740137/111461429-7dd75f00-871d-11eb-90ea-52c5777a8e5e.png)

> * Having that parameter, what is the point of ignoring "xxxxxxxxxx", for example? I mean, I've declared my fair share of "ii" and "zz"... variables. But is there any language where it is common to declare a variable such as "xxxxxx" or "sssss"? (Maybe there is, I'm actually asking)

The goal of the spell checker is to be strict enough to be useful, but not so strict to be annoying. One of the biggest complaints is that it shows too many errors. If someone types `xxxxxxxx` they most likely did it on purpose.

Even so, I agree that it should be a configurable option if wanted. It is currently hard coded into the `cspell` tool. Please feel free to open an issue: [Issues · cspell](https://github.com/streetsidesoftware/cspell/issues)

**Comment by salgiza** [comment]
Thanks a lot for your answer, I didn't know about using overrides to set different behaviours based on the file type (I should have checked the docs before asking, sorry 🤦‍♂️ ).

I'm closing the ticket, as it is clear that the spell-checker has everything covered!

**Comment by Jason3S** [comment]
No worries. Maybe someone else will find this discussion useful.
